### PR TITLE
feat: MinIO Deployment and S3 to S3 tutorial

### LINKED ISSUE #34: Data transfer: extend samples to utilize S3 (minio)
## WHAT

The MXD walkthrough should be extended so that one provider connector, e.g. "Alice" hosts an asset, that is actually stored in S3 (using the MinIO container image). 
Bob then transfers the data into some arbitrary location, e.g. his own S3 store.

In the course of this issue, it should also be tested, whether a transfer HTTP -> S3 can be achieved.

## WHY
Highlighting standard Tx-EDC use cases and making them accessible for users of MXD.

## HOW
- [ ] add terraform resources for MinIO (_suggestion_: use a `kubernetes_deployment`)
- [ ] create a bucket on provider side and put some test data into that bucket. This needs to be done automatically, without manual commands. (_suggestion_: could use a `job`)
- [ ] add relevant documentation to `mxd/README.md` outlining the process, including the correct `curl` command and transfer request JSON
- [ ] repeat for transfers `HTTP -> S3`
- [ ] Suggestion: after each transfer, show the contents of the relevant database tables on provider and consumer side, to further illustrate what's going on.

### PR DESCRIPTION
**Pull request by hemantxpatel**
## Description

- MinIO Deployment
- Seed jobs to create bucket / upload test document.
- S3 to S3 Tutorial

Closes #34 
## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: mxd/modules/connector/main.tf [hunk 1]
```diff
@@ -62,6 +69,13 @@ resource "helm_release" "connector" {
           }
         }
       }
+      dataplane : {
+        aws : {
+          endpointOverride : "http://${local.minio-url}"
+          accessKeyId : var.minio-config.minio-username
```
[the remark this answers is not in the document]
Comment (hemantxpatel) [reply mxd/modules/connector/main.tf]: Added secrets in Vault with standard edc defined keys `edc.aws.access.key` \ `edc.aws.secret.access.key`.
`S3` extension will pick secrets from Vault.

File: mxd/modules/connector/main.tf [hunk 2]
```diff
@@ -37,7 +44,7 @@ resource "helm_release" "connector" {
           "postStart" : [
             "sh",
             "-c",
-            "sleep 5 && /bin/vault kv put secret/client-secret content=${local.client_secret} && /bin/vault kv put secret/aes-keys content=${local.aes_key_b64} && /bin/vault kv put secret/${var.ssi-config.oauth-secretalias} content=${var.ssi-config.oauth-clientsecret}"
+            "sleep 5 && /bin/vault kv put secret/client-secret content=${local.client_secret} && /bin/vault kv put secret/aes-keys content=${local.aes_key_b64} && /bin/vault kv put secret/${var.ssi-config.oauth-secretalias} content=${var.ssi-config.oauth-clientsecret} && /bin/vault kv put secret/${var.minio-config.minio-username}-alias content='${local.minio-secret}' "
```
[the remark this answers is not in the document]
Comment (hemantxpatel) [reply mxd/modules/connector/main.tf]: Used  terraform `join(separator, list)` method.

File: mxd/docs/File Transfer S3 to S3.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+    "bucketName": "bob-bucket"
+  },
+  "connectorId": "BPNL000000000001",
+  "@context": {}
+}
+```
+> Please note, `state` should be `COMPLETED`. If it is `TERMINATED`, it means, transfer has been failed. Please check logs for more details.
+
+## 13. Validate Transfer
+Now, we can validate whether file has been transferred in `bob-bucket` or not.
```
Comment (paullatzelsperger) [inline mxd/docs/File Transfer S3 to S3.md:433]:
```suggestion
Now we can validate that file has been transferred to `bob-bucket`.  
```

File: mxd/docs/File Transfer S3 to S3.md [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+  },
+  "connectorId": "BPNL000000000001",
+  "@context": {}
+}
+```
+> Please note, `state` should be `COMPLETED`. If it is `TERMINATED`, it means, transfer has been failed. Please check logs for more details.
+
+## 13. Validate Transfer
+Now, we can validate whether file has been transferred in `bob-bucket` or not.  
+Once transfer is complete, file `alice-test-document.txt` should be present in `bob-bucket`.
```
Comment (paullatzelsperger) [inline mxd/docs/File Transfer S3 to S3.md:434]:
```suggestion
Once transfer is complete, `alice-test-document.txt` should be present in `bob-bucket`.
```

**Review by paullatzelsperger — COMMENTED** [review]
will merge once minor nits are resolved

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**